vim-patch:8.2.3598,3599,3600: some filetypes are not recognized

clason opened the pull request:
Problem:    RouterOS filetype is not recognized.
Solution:   Add file and script patterns. (closes vim/vim#9097)
https://github.com/vim/vim/commit/0818ab82e7058145366ebbe759f0b3f74724bdfd

Problem:    Not all gdbinit files are recognized.
Solution:   Add "gdbinit". (Doug Kearns)
https://github.com/vim/vim/commit/782b4bbc163e03ebe98d25bc62b9d82cba8f91a1

Problem:    Filetype test fails.
Solution:   Add missint change.
https://github.com/vim/vim/commit/314b773abbb9b1ce0020d83482c6daf7ad6a42a2

clason committed:
vim-patch:8.2.3598: RouterOS filetype is not recognized

Problem:    RouterOS filetype is not recognized.
Solution:   Add file and script patterns. (closes vim/vim#9097)
https://github.com/vim/vim/commit/0818ab82e7058145366ebbe759f0b3f74724bdfd

clason commented:
Hmm... New test fails with
```
function RunTheTest[37]..Test_script_detection[1]..Run_script_detection line 6: for text: ['#!/path/rsc']: Expected 'routeros' but got 'conf'
```
also upstream, so let's wait how Bram deals with it.

zeertzjq commented:
The tests probably depend on a pending runtime update.

clason commented:
Let's see how long that takes, then ;)

seandewar reviewed (approved):
LGTM, just needs the rt update for the script check

clason committed:
vim-patch:8.2.3599: not all gdbinit files are recognized

Problem:    Not all gdbinit files are recognized.
Solution:   Add "gdbinit". (Doug Kearns)
https://github.com/vim/vim/commit/782b4bbc163e03ebe98d25bc62b9d82cba8f91a1

clason commented:
might as well add more related patches while we wait...

zeertzjq commented:
<https://github.com/vim/vim/commit/314b773abbb9b1ce0020d83482c6daf7ad6a42a2>

clason committed:
vim-patch:8.2.3600: filetype test fails

Problem:    Filetype test fails.
Solution:   Add missint change.
https://github.com/vim/vim/commit/314b773abbb9b1ce0020d83482c6daf7ad6a42a2

clason commented:
Flakers gonna flake; I'll go ahead and merge.
